src/main.rs

```diff
@@ -2114,7 +2120,36 @@ extern "C" fn time_ticks_per_second() -> common::Ticks {
 	common::Ticks(1_000_000)
 }
 
-static IRQ_PIN: NeoMutex<Option<IrqPin>> = NeoMutex::new(None);
+/// Power the system off.
+extern "C" fn power_off() -> ! {
+	todo!();
+}
+
+/// Reboot the system.
+extern "C" fn power_reboot() -> ! {
+	todo!();
+}
+
+/// Performs a compare-and-swap on `value`.
+///
+/// * If `value == old_value`, sets `value = new_value` and returns `true`
+/// * If `value != old_value`, returns `false`
+extern "C" fn compare_and_swap_bool(value: &AtomicBool, old_value: bool, new_value: bool) -> bool {
+	let state = unsafe { critical_section::acquire() };
```

Comment:
Why don't you use `critical_section::with`, which would not need `unsafe` blocks?

Reply:
TIL!